pkg/db/tx.go

```diff
@@ -128,18 +120,28 @@ func (d *dbTx) Exec(query string, args ...interface{}) (pgconn.CommandTag, error
 		"query": queryToString(query),
 		"took":  time.Since(start),
 	})
-	if isUniqueViolation(err) {
-		return nil, ErrAlreadyExists
-	}
 	if err != nil {
-		dbErrorsCounter.WithLabelValues("exec").Inc()
-		log.WithError(err).Error("SQL query failed with error")
-		return res, err
+		return res, d.handleSQLError(err, "exec", query)
 	}
 	log.Trace("SQL query executed successfully")
 	return res, err
 }
 
+func (d *dbTx) handleSQLError(err error, cmdType string, query string) error {
+	d.logger.Trace("SQL operation err:", err)
```

Comment:
- Use `.WithError(err)` to capture `err` in a structure way - prefer to pass fields to the log in this format as later when we process the logs we can extract the information easily.
- Previously we used the scoped `log` variable which captures cmd type and query - we need to log this information here and in the error too, when we switched to use the receiver.